app/lib/screens/home/home.dart

```diff
@@ -1,37 +1,23 @@
-import 'package:app/navbar/navigation.dart';
-import 'package:app/sidebar/sidebar.dart';
+import 'package:app/navigations/nav_bar/navigation.dart';
+import 'package:app/navigations/side_bar/sidebar.dart';
 import 'package:flutter/material.dart';
+import 'package:app/providers/navigation_provider.dart';
+import 'package:provider/provider.dart';
 
-class HomePage extends StatelessWidget {
+class HomeScreen extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
+    print('Application starting....');
     return Scaffold(
+      appBar: AppBar(),
       drawer: SideBar(),
-      appBar: AppBar(
-        title: Text('Home Page'),
+      body: Consumer<NavigationProvider>(
+        builder: (context, model, _) => IndexedStack(
+          index: model.getIndex,
+          children: model.getCurrentScreen,
+        ),
       ),
-      body: Center(
-          child: Column(
-        mainAxisAlignment: MainAxisAlignment.center,
-        children: [
-          Text('This is Home Page'),
-          SizedBox(
-            height: 10.0,
-          ),
-          RaisedButton.icon(
-            onPressed: () {
-              Navigator.push(
-                context,
-                MaterialPageRoute(
-                  builder: (context) => NavigationBarController(),
-                ),
-              );
-            },
-            icon: Icon(Icons.add_circle_outline),
-            label: Text('New Travel'),
-          ),
-        ],
-      )),
+      bottomNavigationBar: BottomNavBar(),
```

Comment:
Good refactoring example. I'm sure you're well aware of the following Clark - but to other readers:
1. We should extract all contextually different widgets/objects out to their own files
1. This adds to reusability and our understanding of composition
1. This supports the preference of composition over inheritance
1. [Flutter architecture is build with this in mind](https://flutter.dev/docs/resources/architectural-overview)

And, as a more personal note: this reduces file length and we like shallow files; not only does it make reading easier but because all contextually different code is separate, testing becomes easier.

Reply:
Also, by extracting this widget it gives me the ability to determine if it's building or not